src/components/List/index.js

```diff
@@ -0,0 +1,137 @@
+import React, { useRef, useState } from 'react';
+import './style.scss';
+import Arrow from '../../assets/images/Arrow-down.svg';
+import { useOnClickOutside } from 'usehooks-ts';
+
+const List = ({ options }) => {
+    const listRef = useRef();
+    const [state, setState] = useState({
+        selected: false,
+        showMenu: false,
+        searchValue: '',
+        inputValue: '',
+        focusedItem: 0,
+    });
+
+    const handleInputClick = (e) => {
+        e.stopPropagation();
+        setState((prevState) => ({
```

Comment:
you don't need to specify the prev-state every time just update the relevant items in the state